### ISSUE TITLE
[NFR] \Phaclon\Db\Index: TYPE

### ISSUE THREAD

**Issue by lantian** [phalcon/cphalcon#1989]
**[NOW]:**
Currently we can create index via db adapter:

```php
$db->addIndex(new Index('name', $columns));
```

But we can't set index type. Index has only two properties (name and columns) - http://docs.phalconphp.com/en/latest/api/Phalcon_Db_Index.html

**[NFR]:**
Adding new fields 'type' to improve index:

```php
class Index {

public function __constructor($name, $columns, $type = null){...}
public function getType(){...}

}
```

This will allow to generate sql: 
```sql
ALTER TABLE `$table` ADD $type INDEX `$name` (...$columns...);
```

**Comment by sergeyklay** [comment]
@andresgutierrez Issue is solved

**Comment by Green-Cat** [comment]
@sergeyklay Implemented in #2449
